yarn-project/aztec-nr/aztec/src/history/public_value_inclusion.nr

```diff
@@ -16,6 +16,50 @@ use crate::{
 };
 
 pub fn prove_public_value_inclusion(
+    value: Field, // The value that we want to prove is in the public data tree
+    storage_slot: Field, // The storage slot in which the value is stored
+    context: PrivateContext
+) {
```

Comment:
We can do the same as above to avoid code duplication.

Reply:
@benesjan, do you think the `contract_address` should be an input instead of using the current context? I'm thinking that having it as an input would be good since we can then handle cases where I want to prove something around historical state of **another** contract.

Reply:
That's a great idea 👍